fastly/secret_store.go

```diff
@@ -379,3 +384,77 @@ func (c *Client) DeleteSecret(i *DeleteSecretInput) error {
 	}
 	return resp.Body.Close()
 }
+
+// ClientKey is an X25519 public key that can be used with
+// golang.org/x/crypto/nacl/box to encrypt secrets locally before
+// uploading them to the Fastly API.  A client key is valid only for a
+// short amount of time, and should be used immediately.  The key is not
+// valid after the ExpiresAt time.
+//
+// Client keys are signed, and the attached signature must be validated
+// using the public signing key before it is used.  A ValidateSignature
+// method is provided for this purpose.
+type ClientKey struct {
+	PublicKey []byte    `json:"public_key"`
+	Signature []byte    `json:"signature"`
+	ExpiresAt time.Time `json:"expires_at"`
+}
+
+// ValidateSignature returns true if the client key's signature for the
+// provided public signing key.
```

Comment:
This sentence feels incomplete.

I had a look at the `ed25519` package to get a sense of what its `Verify` does and came up with the following:

```suggestion
// ValidateSignature reports if signingKey was used to 
// generate the signature. It will panic if len(publicKey) 
// is not ed25519.PublicKeySize.
// https://pkg.go.dev/crypto/ed25519@go1.19.5#PublicKeySize
```